docs/adrs/versioning-strategy.md

```diff
@@ -0,0 +1,45 @@
+# ADR: Versioning strategy for containers and code packages
+
+## Context
+
+Currently, all containers and code packages are versioned with the full git commit SHA.  Whilst this tells us exactly which commit a package has been built from, it doesn't make the version very accessible to consumers of the package without looking into it further.
+
+In some places, we have implemented a crude semantic version style where the version is 1.0.<github_run_number>. However, this does not allow us to easily bump major/minor versions or highlight breaking changes.
+
+## Decision
+
+1. Use the [GitVersion GitHub Action](https://github.com/marketplace/actions/gittools) to implement semantic versions across all containers and applications: https://gitversion.net/docs/learn/intro-to-semver
+2. Packages from `main` branch are to have a full semantic version like this: `1.2.3`
+3. Packages from non-`main` branches are to have a full semantic version like this: `1.2.3-pr17.12`
+   * `pr17` - denoting that the package is an output of a (hypothetical) PR #17 
+   * `.12` - number of pushes in made to the branch/PR
+   * The above values are default out-of-the-box functionality of the GitVersion tool mentioned above
+4. For consistency, the docker image tags should also use the same versioning strategy
```

Comment:
For docker images there should be some additional conditions. One image should have 3 version tags: 
- one with major number (movable)
- one with major.minor (movable)
- one with major.minor.patch
Image tag from only major number is used on latest build. Tag with major and minor number is used on latest build with that major and minor number. For example docker image with id `7e43740c028d` is build from version `1.2.3` and gets tags `1`, `1.2` and `1.2.3`. The next build creates image with id `836e499cd934` from version `1.3.2` and gets tags `1`, `1.3` and `1.3.2`
This way it would be easier to update system and give the option to hang on to specific version if needed.

Reply:
This has been addressed in this PR: https://github.com/buerokratt/Cross-functional-requirements/pull/95

I'll resolve this comment and any additional discussion/feedback on this will continue on the new PR. Thanks dude!